### PR TITLE
vxWorks: implement get_path() and get_mode() for File fmt::Debug

### PR DESCRIPTION
**Pull request by BaoshanPang**

### CODE REVIEW

File: src/libstd/sys/vxworks/fs.rs [hunk 1]
```diff
@@ -400,13 +400,28 @@ impl FromInner<c_int> for File {
...
-            None
+        fn get_path(fd: c_int) -> Option<PathBuf> {
+            let mut buf = vec![0;libc::PATH_MAX as usize];
+            let n = unsafe { libc::ioctl(fd, libc::FIOGETNAME, buf.as_ptr()) };
+            if n == -1 {
+                return None;
+            }
+            let l = buf.iter().position(|&c| c == 0).unwrap();
+            buf.truncate(l as usize);
+            buf.shrink_to_fit();
```
Comment (nagisa) [inline src/libstd/sys/vxworks/fs.rs]: Don’t need to `shrink_to_fit` here as the `PathBuf` is formatted in immediately after, this basically wastes time making a roundtrip to the allocator for no real benefit.

File: src/libstd/sys/vxworks/fs.rs [hunk 2]
```diff
@@ -400,13 +400,27 @@ impl FromInner<c_int> for File {
...
-        fn get_path(_fd: c_int) -> Option<PathBuf> {
-            // FIXME(#:(): implement this for VxWorks
-            None
+        fn get_path(fd: c_int) -> Option<PathBuf> {
+            let mut buf = vec![0;libc::PATH_MAX as usize];
+            let n = unsafe { libc::ioctl(fd, libc::FIOGETNAME, buf.as_ptr()) };
+            if n == -1 {
+                return None;
+            }
+            let l = buf.iter().position(|&c| c == 0).unwrap();
```
Comment (kennytm) [inline src/libstd/sys/vxworks/fs.rs]: I don't know about vxWorks, but is it guaranteed that FIOGETNAME's result is always shorter than PATH_MAX (1024 bytes) including the trailing `\0`?

Comment (BaoshanPang) [reply src/libstd/sys/vxworks/fs.rs]: In vxWorks, we are using strlcopy to fill the buffer(arg) like this:
strlcpy ((char *) arg, name, PATH_MAX);

**Pull request merged**